newsrevenuehub/rev-engine
# [NRH-263] BenefitLevels Updates

#### What's this PR do?
Implements the backend changes required to make BenefitLevels behave as expected:

- Removes models DonorBenefits, BenefitTier
- Adds BenefitLevel with M2M on RevenueProgram. BenefitLevel is orderable by "level" field on through table
- Makes Benefit orderable by "order" field on through table.
- Serializes benefit_levels on to donation_page

#### How should this be manually tested?
This PR does not implement the donation page facing changes.  That's NRH-283. This can be tested in Django admin as follows:

1. Observe that DonorBenefits, BenefitTier, and Benefit are no longer under the "Donation pages" app
2. Observe that "Benefit levels" and "Benefits" appear under the "Organizations" app
3. Create a new, or edit an existing revenue program.
4. At the bottom of the edit view, notice the inline interface for associating benefit levels with a revenue program.
5. From here you can link existing levels and order them, or create new levels (by clicking the green plus).
6. When creating a new BenefitLevel, observe that a similar interface exists for adding a Benefit to a BenefitLevel.
7. Note that you cannot currently test the outgoing serialized representation of the benefit levels on a donation page. This will be testable in NRH-283

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?
No
